### PR TITLE
Build with 2.13

### PR DESCRIPTION
**Pull request by crispywalrus**
this required finagle 20.1.0 which has now been released!

### CODE REVIEW

**Comment by crispywalrus** [comment]
@adriancole I'm not sure if it's possible to publish off this branch with the same version number as the 2.12 branch. It would be practical to publish something like `2.2.1` I can't find any way to make maven do a matrix build with multiple versions of scala so I feel like that's the best I can do for now.

**Comment by adriancole** [comment]
bintray doesn't let you publish again with the same version number. However, sonatype directly allows this. The hack I did once was to delete the version in bintray, then publish again :P

skipping bintray to use sonatype directly is ok, except you need to use GPG to sign. Right now bintray automates the signing part. When bintray breaks, I just publish manually after entering my GPG password locally.

ex zipkin-classic is completely whacked from a bintray point of view, so I can't publish with the normal automated tag thing. What I do is just use old school sonatype with this profile

https://github.com/openzipkin/zipkin-classic/blob/master/pom.xml#L178-L204
https://github.com/openzipkin/zipkin-classic/issues/14

in other words if we want to do two releases, one would have to be done manually, or we do some script that does them both and overrides the constraints somehow

cc @openzipkin/devops-tooling

**Comment by crispywalrus** [comment]
the artifacts would be different, with a `_2.13` on the artifacts.

**Comment by adriancole** [comment]
@crispywalrus ack on that.. it is just technically the version in bintray would be the same. https://bintray.com/openzipkin/maven/zipkin-finagle unless the build was parameterized to make a different bintray app for scala 2.13. This is possible to do, just some "paperwork" in bintray to make something like https://bintray.com/openzipkin/maven/zipkin-finagle_2.13

**Comment by adriancole** [comment]
bintray may or may not allow us to publish multiple apps to the same maven coordinates.. this may limit the approach to use a different app name to address the dual publish thing. Someone at bintray could clarify this with an email to their support

**Comment by jbripley** [comment]
Any updates on publishing Scala 2.13 artefacts?

**Comment by adriancole** [comment]
I added https://bintray.com/openzipkin/maven/zipkin-finagle_2.13 and also added @crispywalrus org perms to bintray

I'd suggest making a branch off master named `scala_2.13` or such and make the changes done here, notably changing the bintray plugin to use `zipkin-finagle_2.13`.

The automatic publish script based on release-N.N.N tag may or may not assume releases are off the master branch. It might need some change not sure. In any case, manual release would work even if the tag based stuff doesn't.  Not sure the RELEASE.md has manual instructions, but here they are https://github.com/openzipkin/brave/blob/master/RELEASE.md#manually-releasing

The first release of bintray has to happen not completely automated as there is chicken-egg. You aren't allowed to request for jcenter until you already have code. So doing one release manually will need to happen in order to get the next one working.

Hope these notes help cc @openzipkin/devops-tooling

**Comment by crispywalrus** [comment]
Ok, so I've now released 2.1.8 for scala 2.12. Next step is going to try a manual release for 2.13.

**Comment by adriancole** [comment]
same compile glitch recently I noticed here https://github.com/openzipkin/zipkin-finagle/pull/88

**Comment by adriancole** [comment]
lets carry on in  #75 as the code changes are in master now. thanks for the help

**Comment by adriancole** [comment]
https://github.com/openzipkin/zipkin-finagle/pull/89 to complete the deploy part (note this doesn't run tests also in scala 2.13 just recompiles and repackages)
